CODE_OF_CONDUCT.md

```diff
@@ -0,0 +1,88 @@
+
+Harassing other users is never tolerated, whether via public or private media.
+
+Avoid using offensive or harassing nicknames, or other identifiers that might detract from a friendly, safe, and welcoming environment for all.
+
+Harassment includes, but is not limited to: any behavior that threatens or demeans another person or group, or produces an unsafe environment; harmful or prejudicial verbal or written comments related to gender, gender expression, gender identity, sexual orientation, disability, ethnicity, religion, age, physical appearance, body size, race, or similar personal characteristics; inappropriate use of nudity, sexual images, and/or sexually explicit language in public spaces; threats of physical or non-physical harm; deliberate intimidation, stalking or following; harassing photography or recording; sustained disruption of talks or other events; inappropriate physical contact; and unwelcome sexual attention.
+
+Media shared through public infrastructure run by the IPFS team must not contain illegal or infringing content. You should only publish content via IPFS public infrastructure if you have the right to do so. This includes complying with all software license agreements or other intellectual property restrictions. You will be solely responsible for any violation of laws or
+others’ intellectual property rights.
+
+## Reporting Violations of this Code of Conduct
+
+If you believe someone is harassing you or has otherwise violated this Code of Conduct, please [contact us](#contact-info) to send us an abuse report. If this is the initial report of a problem, please include as much detail as possible. It is easiest for us to address issues when we have more context.
+
+## Consequences
+
+Unacceptable behavior from any community member will not be tolerated.
+
+Anyone asked to stop unacceptable behavior is expected to comply immediately.
+
+If a community member engages in unacceptable behavior, the team may take any action they deem appropriate, up to and including a temporary ban or permanent expulsion from the community without warning (and without refund in the case of a paid event or service).
+
+## Addressing Grievances
+
+If you feel you have been falsely or unfairly accused of violating this Code of Conduct, you should notify the team. We will do our best to ensure that your grievance is handled appropriately.
+
+In general, we will choose the course of action that we judge as being most in the interest of fostering a safe and friendly community.
+
+On IRC, let one of the ops know if you think that someone has transgressed against the Code of Conduct. If you would like to be an op and promise to help maintain and abide by the code, please let us know.
+
+## Contact Info
+
+The Drand team currently defers to the IPFS abuse team for Code of Conduct resolutions. Please contact abuse@ipfs.io if you need to report a problem or address a grievance related to an abuse report.
```

Comment:
Just have an abuse@drand? I feel we should try to prioritise getting help in this document not who is providing it. Personally I’d remove references to IPFS here.

Reply:
IPFS is large enough to have trained an abuse team to handle these issues, and can act as a neutral 3rd party arbiter. we could do much worse than deferring to that abuse team if us an them are willing to have that happen.

Reply:
when there are code of conduct issues, it's important that the person sending the report knows who it's going to, and has transparency in that. Ideally we list the people here, but if we defer to IPFS i think it's okay to include a link to the IPFS code of conduct so that there's one authoritative source of that membership rather than this one potentially getting stale. (re your pointer to credit below)

Reply:
> IPFS is large enough to have trained an abuse team to handle these issues, and can act as a neutral 3rd party arbiter. we could do much worse than deferring to that abuse team if us an them are willing to have that happen.

Um, okay. Just to be clear I was not suggesting that it’s a bad idea to defer this to the IPFS abuse team or that they are in any way a bad choice for it.

My thoughts were just to ensure the contact info is front and centre, instead of an advertisement for IPFS, which is how I felt it could come across.

In hindsight I think you’re right and if we’re going to defer to IPFS for abuse issues then for transparency we actually _shouldn’t_ have our own email address.

In which case we can easily resolve this by switching the sentence around so that the contact details come first, followed by clarification of who is receiving it and why.